044-cluster-wide-volume-usage-quota-management.md

```diff
@@ -0,0 +1,342 @@
+2. throttle factor recalculation at minute 1 failed, continue using throttle factor 0.6
+3. throttle factor recalculation at minute 2 failed, continue using throttle factor 0.6
+4. throttle factor recalculation at minute 3 failed, use throttle factor fallback of 1.0
+5. throttle factor recalculation at minute 4 succeeded, calculated throttle factor 0.6 and use it
+
+## Configuration Summary
+
+|                                                                | type   | default | valid values      |                                                                                                                                  |
+|----------------------------------------------------------------|--------|---------|-------------------|----------------------------------------------------------------------------------------------------------------------------------|
+| client.quota.callback.static.storage.volume.source             | string | local   | local, cluster    | set to cluster to obtain volume descriptions from cluster, see [VolumeSource](#volume-source)                                    |
+| client.quota.callback.static.storage.check-interval            | long   | 0       | (0, ...)          | 0 means disabled, otherwise this is the milliseconds between polling to describe the cluster (this is an existing configuration) |
+| client.quota.callback.static.kafka.admin.bootstrap.servers     | string | null    |                   | required if volume source is cluster, bootstrap.servers for [admin client](#admin-client-configuration) used to get cluster data |
+| client.quota.callback.static.kafka.admin.*                     | ?      |         |                   | optionally users can configure arbitrary properties of the [admin client config](#admin-client-configuration)                    |
+| client.quota.callback.static.throttle.factor.fallback          | double | 1.0     | (0.0, 1.0)        | sets the [Throttle Factor Fallback](#throttle-factor-fallback)                                                                   |
+| client.quota.callback.static.throttle.factor.validity-duration | string | PT5M    | ISO8601 durations | sets the [Throttle Factor Validity Duration](#throttle-factor-validity-duration)                                                 |
+
+### Limit Configuration
+When using cluster sourced volumes the user must configure a single [limit type](#limit-type-configuration), all are incompatible with `local` volume source
+
+|                                                                         | type   | default | valid values |                                                         |
+|-------------------------------------------------------------------------|--------|---------|--------------|---------------------------------------------------------|
+| client.quota.callback.static.storage.perVolumeLimit.availableBytesBelow | long   | null    | (0, ...)     | stop message production if availableBytes <= this value |
```

Comment:
Is it common to use camel case in configuration names instead of dots?

Reply:
What about `min.available.bytes` instead? That would line up nicely with Kafka configs such as `min.insync.replicas` or `fetch.min.bytes`.
Same for the other config below.

Reply:
sounds good, have switched from `below` to `min` prefix.